src/skins/vector/views/organisms/ViewSource.js

```diff
@@ -21,6 +21,22 @@ var React = require('react');
 module.exports = React.createClass({
     displayName: 'ViewSource',
 
+    componentDidMount: function() {
+        document.addEventListener("keydown", this.onKeyDown);
+    },
+
+    componentWillUnmount: function() {
+        document.removeEventListener("keydown", this.onKeyDown);
+    },
+
+    onKeyDown: function(ev) {
+        if (ev.keyCode == 27) { // escape
+            ev.stopPropagation();
+            ev.preventDefault();
+            this.props.onFinished();
```

Comment:
This is an ongoing problem I have with react/vector split which makes this line here *really* non-obvious. This will crash if `onFinished` is not passed as a prop (undefined is not a function). The class isn't saying it accepts this prop. There are no default props. React SDK *may* be supplying the prop (it isn't in this case). I have no idea **what** uses this at all, which makes refactoring Really Hard. I would prefer that:
 - The props the class uses are explicitly defined.
 - There are default prop implementations for optional props.

This in practice means adding:

```javascript
module.exports = React.createClass({
 [...]

  propTypes: {
    onFinished: React.PropTypes.func // adding .isRequired if this is a required prop in which case don't bother with a default prop
  },

  getDefaultProps: function() {
    return {
      onFinished: function() {} // nop
    }
  },

  [...]
}
```

We don't do this currently everywhere, but I would really like to be in a place where we can have some confidence that certain methods exist.